### ISSUE TITLE
[CI] Windows packaging tests failing to rename log file while in use

### ISSUE THREAD

**Issue by DaveCTurner** [elastic/elasticsearch#86978]
**Build scan:**
https://gradle-enterprise.elastic.co/s/a7qdx3wu2mkuu/tests/:qa:os:destructiveDistroTest.default-windows-archive/org.elasticsearch.packaging.test.ArchiveTests/test60StartAndStop

**Reproduction line:**
`null`

**Applicable branches:**
master

**Reproduces locally?:**
Didn't try

**Failure history:**
https://gradle-enterprise.elastic.co/scans/tests?tests.container=org.elasticsearch.packaging.test.ArchiveTests&tests.test=test60StartAndStop

**Failure excerpt:**
```
org.elasticsearch.packaging.test.ArchiveTests > test60StartAndStop FAILED	
    org.elasticsearch.packaging.util.Shell$ShellException: Command was not successful: [powershell.exe -Command $processInfo = New-Object System.Diagnostics.ProcessStartInfo; $processInfo.FileName = 'C:\tmp\elasticsearch\bin\elasticsearch.bat'; $processInfo.Arguments = '-v -p C:\tmp\elasticsearch\elasticsearch.pid'; $processInfo.RedirectStandardOutput = $true; $processInfo.RedirectStandardError = $true; $processInfo.RedirectStandardInput = $true; $processInfo.UseShellExecute = $false; $process = New-Object System.Diagnostics.Process; $process.StartInfo = $processInfo; $outScript = { $EventArgs.Data | Out-File -Encoding UTF8 -Append 'C:\tmp\elasticsearch\logs\output.out' }; $errScript = { $EventArgs.Data | Out-File -Encoding UTF8 -Append 'C:\tmp\elasticsearch\logs\output.err' }; $stdOutEvent = Register-ObjectEvent -InputObject $process -Action $outScript -EventName 'OutputDataReceived'; $stdErrEvent = Register-ObjectEvent -InputObject $process -Action $errScript -EventName 'ErrorDataReceived'; $process.Start() | Out-Null; $process.BeginOutputReadLine(); $process.BeginErrorReadLine(); $process.StandardInput.WriteLine('null'); Wait-Process -Timeout 25 -Id $process.Id; $process.Id;]	
       result: exitCode = [2] stdout = [7824] stderr = [Wait-Process : This command stopped operation because process "cmd (7824)" is not stopped in the specified time-out.	
    At line:1 char:1031	
    + ... t.WriteLine('null'); Wait-Process -Timeout 25 -Id $process.Id; $proce ...	
    +                          ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~	
        + CategoryInfo          : CloseError: (System.Diagnostics.Process (cmd):Process) [Wait-Process], TimeoutException	
        + FullyQualifiedErrorId : ProcessNotTerminated,Microsoft.PowerShell.Commands.WaitProcessCommand]	
        at app//org.elasticsearch.packaging.util.Shell.runScript(Shell.java:143)	
        at app//org.elasticsearch.packaging.util.Shell.run(Shell.java:73)	
        at app//org.elasticsearch.packaging.util.Archives.runElasticsearchStartCommand(Archives.java:344)	
        at app//org.elasticsearch.packaging.test.PackagingTestCase.runElasticsearchStartCommand(PackagingTestCase.java:333)	
        at app//org.elasticsearch.packaging.test.PackagingTestCase.startElasticsearch(PackagingTestCase.java:398)	
        at app//org.elasticsearch.packaging.test.ArchiveTests.test60StartAndStop(ArchiveTests.java:274)	
    java.lang.RuntimeException: java.nio.file.FileSystemException: C:\tmp\elasticsearch\logs\elasticsearch.log -> C:\tmp\elasticsearch\logs\ArchiveTests.test60StartAndStop.elasticsearch.log: The process cannot access the file because it is being used by another process	
        at org.elasticsearch.packaging.util.FileUtils.mv(FileUtils.java:136)	
        at org.elasticsearch.packaging.test.PackagingTestCase.teardown(PackagingTestCase.java:214)	
        at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:104)	
        at java.base/java.lang.reflect.Method.invoke(Method.java:577)	
        at com.carrotsearch.randomizedtesting.RandomizedRunner.invoke(RandomizedRunner.java:1758)	
        at com.carrotsearch.randomizedtesting.RandomizedRunner$10.evaluate(RandomizedRunner.java:1004)	
        at org.junit.rules.TestWatcher$1.evaluate(TestWatcher.java:55)	
        at org.junit.rules.TestWatcher$1.evaluate(TestWatcher.java:55)	
        at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)	
        at com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:375)	
        at com.carrotsearch.randomizedtesting.ThreadLeakControl.lambda$forkTimeoutingTask$0(ThreadLeakControl.java:831)	
        at java.base/java.lang.Thread.run(Thread.java:833)	
        Caused by:	
        java.nio.file.FileSystemException: C:\tmp\elasticsearch\logs\elasticsearch.log -> C:\tmp\elasticsearch\logs\ArchiveTests.test60StartAndStop.elasticsearch.log: The process cannot access the file because it is being used by another process	
            at java.base/sun.nio.fs.WindowsException.translateToIOException(WindowsException.java:92)	
            at java.base/sun.nio.fs.WindowsException.rethrowAsIOException(WindowsException.java:103)	
            at java.base/sun.nio.fs.WindowsFileCopy.move(WindowsFileCopy.java:403)	
            at java.base/sun.nio.fs.WindowsFileSystemProvider.move(WindowsFileSystemProvider.java:293)	
            at java.base/java.nio.file.Files.move(Files.java:1432)	
            at org.elasticsearch.packaging.util.FileUtils.mv(FileUtils.java:134)	
            ... 11 more	
```

**Comment by rjernst** [comment]
I'm not sure what happened here, but running the windows process in the background was historically difficult, and we have some crazy powershell code to do that, which is another layer that can get out of sync/fail. Now that we have -d on windows, I think we can [get rid of that intermediate layer](https://github.com/elastic/elasticsearch/pull/86995). Ultimately the issue here is Elasticsearch must still be running. Hopefully removing that layer will help us determine why when this occurs again.

**Comment by ldematte** [comment]
After work done by Ryan in https://github.com/elastic/elasticsearch/pull/86995 this issue is no longer valid: the file move exception in cleanup can of course still happen, but the root cause (something strange happening in the powershell script, causing it to timeout waiting for elasticsearch to start) has either been eliminated when getting rid of the powershell script or moved - future failures will look different.

**Comment by ldematte** [comment]
I have examined the build scans and failures for this test class for the last month. This particular test (`test60StartAndStop`) never failed. Other tests in the suite failed randomly with the same error, but in any instance I looked at it was due to builds on older versions (backports) that did not include https://github.com/elastic/elasticsearch/pull/86995 yet or genuine failures due to unrelated work. I'm closing it, but happy to re-open if this happens again.
